application/controllers/AuthenticationController.php

```diff
@@ -35,11 +38,19 @@ public function loginAction()
             $this->redirectNow(Url::fromPath('setup'));
         }
         $form = new LoginForm();
+        if (RememberMe::hasCookie()) {
+            if (! RememberMe::fromCookie()->authenticate()) {
+                $this->getResponse()->setCookie(
+                    RememberMe::unsetCookie()
+                );
+                (new RememberMe())->remove($this->Auth()->getUser()->getUsername());
+            }
+        }
         if ($this->Auth()->isAuthenticated()) {
             // Call provided AuthenticationHook(s) when login action is called
             // but icinga web user is already authenticated
             AuthenticationHook::triggerLogin($this->Auth()->getUser());
-            $this->redirectNow($form->getRedirectUrl());
+            $this->redirectNow($this->params->get('redirect'));
```

Comment:
The default redirect to `dashboard` is missing now if there is no redirect parameter given. Please have look how to get the default redirect URL from the `LoginForm` and how it is passed to `UrlParams::get()`.